chain/ethereum/src/ethereum_adapter.rs

```diff
@@ -288,68 +283,65 @@ where
             true => (to - from).min(*MAX_EVENT_ONLY_RANGE - 1),
         };
 
-        stream::unfold((from, step), move |(start, step)| {
-            if start > to {
-                return None;
-            }
+        // Typically this will loop only once and fetch the entire range in one request. But if the
+        // node returns an error that signifies the request is to heavy to process, the range will
+        // be broken down to smaller steps.
+        futures03::stream::try_unfold((from, step), move |(start, step)| {
+            let logger = logger.clone();
+            let filter = filter.clone();
+            let eth = eth.clone();
+            let subgraph_metrics = subgraph_metrics.clone();
 
-            // Make as many parallel requests of size `step` as necessary,
-            // respecting `LOG_STREAM_PARALLEL_CHUNKS`.
-            let mut chunk_futures = vec![];
-            let mut low = start;
-            for _ in 0..*LOG_STREAM_PARALLEL_CHUNKS {
```

Comment:
What's the reasoning behind dropping parallel chunks?

Reply:
My rationale is on the second paragraph of the PR description.